### DISCUSSION TITLE
Change the surefire/failsafe plugin and use JUnit as Platform

### DISCUSSION THREAD

**Discussion by khmarbaise** [support-and-care/maven-support-and-care#32] [Ideas for Funding]
Currently surefire/failsafe plugin supports the usage of different unit testing frameworks like:
* JUnit 3
* JUnit 4
* TestNG
* JUnit Jupiter (aka JUnit 5)

It might be an idea to reconsider the idea of the surefire/failsafe plugin to handle this cases within a plugin. The JUnit Jupiter Platform supports that out of the box. That means via JUnit Jupiter you can run JUnit 3 and JUnit 4 (via vintage-engine), TestNG via (https://github.com/junit-team/testng-engine) and of course JUnit Jupiter itself.

From my perspective the whole surefire/failsafe plugin could be done much simpler and use the approaches which is already contained in the JUnit Jupiter which would be in line with the IDE's (IntelliJ, Eclipse, Netbean, Visual Studio Code etc.) and other tools like this. That would also simplify the configuration (for example for multi threaded, selecting classes etc.)

That would simplify the code of surefire/failsafe dramatically in many aspects and makes it easier to develop it. A good example of this approache is the https://github.com/sormuras/junit-platform-maven-plugin

**Comment by rmannibucau** [comment]
+1, ultimately it could be replaced - as compiler, javadoc and others, by just a generic toolprovider launcher mojo.

**Comment by ascheman** [comment]
This discussion has been converted to Epic #51 for tracking.

Closing this discussion.
